packages/react/src/lib/factories.ts

```diff
@@ -8,18 +8,19 @@ import {
   ShorthandRenderCallback,
   ShorthandRenderFunction,
   ShorthandRenderer,
+  WithAsProp,
 } from '../types'
 import { mergeStyles } from './mergeThemes'
 
 type HTMLTag = 'iframe' | 'img' | 'input'
 type ShorthandProp = 'children' | 'src' | 'type'
 
-interface CreateShorthandOptions {
+interface CreateShorthandOptions<P> {
   /** Default props object */
-  defaultProps?: Props
+  defaultProps?: P
```

Comment:
```suggestion
  defaultProps?: Partial<Props<P>>
```

- `Partial`: We don't need to specify all default props
- `Props<P>`: We should be able to specify additional props